lib/passport-saml/saml.js

```diff
@@ -792,7 +793,7 @@ SAML.prototype.hasValidSignatureForRedirect = function (samlMessageType, contain
     return this.certsToCheck()
       .then(function(certs) {
         var hasValidQuerySignature = certs.some(function (cert) {
-          return validateSignatureForRedirect(
+          return self.validateSignatureForRedirect(
             urlString, signature, container.SigAlg, cert
```

Comment:
Instead of using `function (cert) {`, you can use an use an arrow function here: `cert => {`. 

Arrow functions do not create a new `this` binding, so you can use `return this.validateSignatureForRedirect` here and it will refer to the outer `this`.